### PR TITLE
upgrade rails to 7.x.x

### PR DESCRIPTION
**Pull request by jibees**
# Todo

- [ ] 
```You have installed oauth2 version 1.4.11, which is EOL.
No further support is anticipated for the 1.4.x series.

OAuth2 version 2 is released.
There are BREAKING changes, but most will not encounter them, and upgrading should be easy!

We have made two other major migrations:
1. master branch renamed to main
2. Github has been replaced with Gitlab

Please see:
• https://gitlab.com/oauth-xx/oauth2#what-is-new-for-v20
• https://gitlab.com/oauth-xx/oauth2/-/blob/main/CHANGELOG.md
• https://groups.google.com/g/oauth-ruby/c/QA_dtrXWXaE

Please upgrade, report issues, and support the project! Thanks, |7eter l-|. l3oling```

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ gem 'activemerchant', '>= 1.78.0'
 gem 'rexml'
 gem 'angular-rails-templates', '>= 0.3.0'
 gem 'awesome_nested_set'
-gem 'ransack', '2.4.2'
+gem 'ransack', '2.5.0'
```
Comment (mkllnk) [inline Gemfile:20]: We should remove this version restraint in a separate PR.

File: Gemfile [hunk 2]
```diff
@@ -79,7 +79,7 @@ gem 'actionpack-action_caching'
 #   AMS is deprecated, we will introduce an alternative at some point
 gem "active_model_serializers", "0.8.4"
 gem 'activerecord-session_store'
-gem 'acts-as-taggable-on', '~> 8.1'
+gem 'acts-as-taggable-on', '~> 9.0'
```
Comment (mkllnk) [inline Gemfile:82]: We should also remove this version and open a PR to upgrade the gem independently.

File: Gemfile [hunk 3]
```diff
@@ -93,7 +93,7 @@ gem 'roadie-rails'
 
 gem 'hiredis'
 gem 'puma'
-gem 'redis', '>= 4.0', require: ['redis', 'redis/connection/hiredis']
+gem 'redis', '>= 4.0'
```
Comment (mkllnk) [inline Gemfile:96]: Can we do this in master already?

**Review by mkllnk — COMMENTED** [review]
Good start. There are some deprecations that need solving first.

**Comment by mkllnk** [comment]
More up-to-date branch: 

- https://github.com/openfoodfoundation/openfoodnetwork/compare/master...mkllnk:openfoodnetwork:rails7

**Comment by jibees** [comment]
This one is very old, and not up to date. Closing then.

**Pull request closed without merging**